src/types.ts

```diff
@@ -203,127 +209,126 @@ export interface SysInfoPretty {
-  resetDisplayedAttributes(): Promise<EnqueuedUpdate>;
-  getAcceptNewFields(): Promise<boolean>;
-  updateAcceptNewFields(acceptNewFields: boolean): Promise<EnqueuedUpdate>;
+  ): Promise<EnqueuedUpdate>
+  resetDisplayedAttributes(): Promise<EnqueuedUpdate>
+  getAcceptNewFields(): Promise<boolean>
+  updateAcceptNewFields(acceptNewFields: boolean): Promise<EnqueuedUpdate>
 }
 
-export interface Meilisearch extends MeiliAxiosErrorInterface {
-  config: Config;
-  getIndex(indexUid: string): Indexes;
-  listIndexes(): Promise<IndexResponse[]>;
-  createIndex(data: IndexRequest): Promise<IndexResponse>;
-  getKeys(): Promise<Keys>;
-  isHealthy(): Promise<boolean>;
-  setHealthy(): Promise<void>;
-  setUnhealthy(): Promise<void>;
-  changeHealthTo(health: boolean): Promise<void>;
-  stats(): Promise<Stats>;
-  version(): Promise<Version>;
-  sysInfo(): Promise<SysInfo>;
-  prettySysInfo(): Promise<SysInfoPretty>;
+export interface Meilisearch extends MeiliSearchApiErrorInterface {
+  config: Config
+  getIndex(indexUid: string): Indexes
+  listIndexes(): Promise<IndexResponse[]>
+  createIndex(data: IndexRequest): Promise<IndexResponse>
+  getKeys(): Promise<Keys>
+  isHealthy(): Promise<boolean>
+  setHealthy(): Promise<void>
+  setUnhealthy(): Promise<void>
+  changeHealthTo(health: boolean): Promise<void>
+  stats(): Promise<Stats>
+  version(): Promise<Version>
+  sysInfo(): Promise<SysInfo>
+  prettySysInfo(): Promise<SysInfoPretty>
 }
 
 export interface MeiliAxiosWrapper {
-  instance: AxiosInstance;
-  cancelTokenSource: CancelTokenSource;
+  instance: AxiosInstance
+  cancelTokenSource: CancelTokenSource
   get<T = any, R = AxiosResponse<T>>(
     url: string,
     config?: AxiosRequestConfig
-  ): Promise<R>;
+  ): Promise<R>
   post<T = any, R = AxiosResponse<T>>(
     url: string,
     data?: any,
     config?: AxiosRequestConfig
-  ): Promise<R>;
+  ): Promise<R>
   put<T = any, R = AxiosResponse<T>>(
     url: string,
     data?: any,
     config?: AxiosRequestConfig
-  ): Promise<R>;
+  ): Promise<R>
   patch<T = any, R = AxiosResponse<T>>(
     url: string,
     data?: any,
     config?: AxiosRequestConfig
-  ): Promise<R>;
+  ): Promise<R>
   delete<T = any, R = AxiosResponse<T>>(
     url: string,
     config?: AxiosRequestConfig
-  ): Promise<R>;
+  ): Promise<R>
 }
 
 /*
  ** ERROR HANDLER
  */
 
-export interface MeiliAxiosErrorInterface extends Error {
+export interface MeiliSearchApiErrorInterface extends Error {
   name: string
   message: string
   stack?: string
 }
-export interface MeiliAxiosErrorResponse {
+export interface MeiliSearchApiErrorResponse {
   status?: number
   statusText?: string
   path?: string
   method?: string
   body?: object
 }
-export interface MeiliAxiosErrorRequest {
+export interface MeiliSearchApiErrorRequest {
   url?: string
   path?: string
   method?: string
 }
 
-export type MeiliAxiosErrorConstructor = new (
+export type MeiliSearchApiErrorConstructor = new (
```

Comment:
I'm not sure to understand, are they "part" of the big error "MeiliSearchApiError"? 🙂 But they are never used/raised, right?

Reply:
They are types! Not actual classes. They help to know javascript users to see what is present in the MeiliSearchApiError object.